src/psyclone/domain/lfric/lfric_extract_driver_creator.py

```diff
@@ -0,0 +1,974 @@
+        providing the arguments. The call will be added to the program and
+        to the symbol table.
+
+        :param program: the PSyIR Routine to which any code must \
+            be added. It also contains the symbol table to be used.
+        :type program: :py:class:`psyclone.psyir.nodes.Routine`
+        :param str name: name of the subroutine to call.
+        :param args: list of all arguments for the call.
+        :type args: list of :py:class:`psyclone.psyir.nodes.Node`
+
+        :raises TypeError: if there is a symbol with the \
+            specified name defined that is not a RoutineSymbol.
+        '''
+        if name in program.symbol_table:
+            routine_symbol = program.symbol_table.lookup(name)
+            if not isinstance(routine_symbol, RoutineSymbol):
+                raise TypeError(
+                    f"Error when adding call: Routine '{name}' is "
+                    f"a symbol of type '{type(routine_symbol).__name__}', "
+                    f"not a 'RoutineSymbol'.")
+        else:
+            routine_symbol = RoutineSymbol(name)
+            program.symbol_table.add(routine_symbol)
+        call = Call.create(routine_symbol, args)
+        program.addchild(call)
+
+    # -------------------------------------------------------------------------
+    def create_output_var_code(self, signature_str, program,
+                               is_input, read_var, postfix,
+                               output_symbols, index=None):
+        # pylint: disable=too-many-arguments, too-many-locals
+        '''
+        This function creates all code required for an output variable.
+        It creates the '_post' variable which stores the correct result
+        from the file, which is read in. If the variable is not also an
+        input variable, the variable itself will be declared (based on
+        the size of the _post variable) and initialised to 0.
+        This function also handles array of fields, which need to get
+        an index number added.
+
+        :param str signature_str: the name of original variable (i.e. \
+            without _post), which will be looked up as a tag in the symbol \
+            table. If index is provided, it is added to the tag name using \
+            ${index}.
+        :param program: the PSyIR Routine to which any code must \
+            be added. It also contains the symbol table to be used.
+        :type program: :py:class:`psyclone.psyir.nodes.Routine`
+        :param boolean is_input: True if this variable is also an input \
+            parameters.
+        :param str read_var: the readvar method to be used including the \
+            name of the PSyData object (e.g. 'psy_data%ReadVar')
+        :param str postfix: the postfix to use for the expected output \
+            values, which are read from the file.
+        :param output_symbols: the list of output symbols for which a check \
```

Comment:
That's just ... bad coding style :) I wanted to avoid to return a pair of symbols from that subroutine (don't ask me why :) ), so instead I passed in a result list from the caller to which the symbol-pair was added. Made absolutely no sense, and is fixed now.